modules/ftrackRtdProvider.js

```diff
@@ -0,0 +1,691 @@
+    s.D9_102 = window.screen ? window.screen.height : undefined;
+    s.D9_103 = window.devicePixelRatio;
+    s.D9_110 = d.getTime();
+    s.D9_111 = d.getTimezoneOffset();
+    s.D9_120 = navigator.platform;
+    s.D9_121 = navigator.language || navigator.browserLanguage;
+    s.D9_122 = navigator.appCodeName;
+    s.D9_123 = navigator.maxTouchPoints || 0;
+    var m = ft.isM(s.D9_120, s.D9_123);
+    s.D9_130 = ft.flashVersion(m);
+    s.D9_131 = ft.acrobatVersion(m);
+    s.D9_132 = ft.silverlightVersion(m);
+    s.D9_133 = ft.getMimeTypes(m);
+    s.D9_134 = ft.getPlugins(m);
+    s.D9_140 = ft.encodeURIComponent(ft.location());
+    s.D9_141 = ft.encodeURIComponent(ft.referrer());
+    s.D9_150 = ft.bh64();
+    s.D9_151 = ft.bh();
+    return s;
+  },
+  initFt: function initFtrack(r) {
+    var ft = {};
+
+    function setResultObject(i, s) {
+      if (i !== undefined && r !== undefined) {
+        r['D9_'.concat(i.toString())] = s;
+      }
+    }
+
+    var FtBh = function FtBh(options) {
+      var nativeForEach, nativeMap;
+      nativeForEach = Array.prototype.forEach;
+      nativeMap = Array.prototype.map;
+
+      this.each = function (obj, iterator, context) {
+        if (obj === null) {
+          return;
+        }
+
+        if (nativeForEach && obj.forEach === nativeForEach) {
+          obj.forEach(iterator, context);
+        } else {
+          if (obj.length === +obj.length) {
+            for (var i = 0, l = obj.length; i < l; i++) {
+              if (iterator.call(context, obj[i], i, obj) === {}) {
+                return;
+              }
+            }
+          } else {
+            for (var key in obj) {
+              if (obj.hasOwnProperty(key)) {
+                if (iterator.call(context, obj[key], key, obj) === {}) {
+                  return;
+                }
+              }
+            }
+          }
+        }
+      };
+
+      this.map = function (obj, iterator, context) {
+        var results = [];
+
+        if (obj == null) {
+          return results;
+        }
+
+        if (nativeMap && obj.map === nativeMap) {
+          return obj.map(iterator, context);
+        }
+
+        this.each(obj, function (value, index, list) {
+          results[results.length] = iterator.call(context, value, index, list);
+        });
+        return results;
+      };
+
+      if (typeof options === 'object') {
+        this.hasher = options.hasher;
+        this.indexProperties = options.indexProperties;
+      } else {
+        if (typeof options === 'function') {
+          this.hasher = options;
+        }
+      }
+    };
+
+    FtBh.prototype = {
+      get: function get() {
+        var ua = navigator.userAgent.toLowerCase();
+        var keys = [];
+        var navLang = navigator.language || navigator.browserLanguage;
+        var navLangArr = navLang.split('-');
+
+        if (typeof navLangArr[0] == 'undefined') {
+          // navLang = navLang;
```

Comment:
Is this intended? You could do

```
if (typeof navLangArr[0] !== 'undefined') {
  navLang = navLangArr[0];
}
```

Reply:
Thanks @Rothalack , I can make this change.

Reply:
Ok, this has been committed.